text/3892-complex-numbers.md

```diff
@@ -0,0 +1,245 @@
+- Feature Name: complex-numbers
+- Start Date: 2025-12-02
+- RFC PR: [rust-lang/rfcs#3892](https://github.com/rust-lang/rfcs/pull/3892)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+## Summary
+[summary]: #summary
+
+FFI-compatible and calling-convention-compatible complex types are to be introduced into `core` to ensure synchronity with C primitives.
+
+## Motivation
+[motivation]: #motivation
+
+The definition of complex numbers in the C99 standard defines the _memory layout_ of a complex number but not its _calling convention_. 
+This makes crates like `num-complex` untenable for calling C FFI functions containing complex numbers without at least a level of indirection (`*const Complex`) or the like.
+Only in `std` is it possible to make an additional repr to match the calling convention that C uses across FFI boundaries. 
+In essence, this RFC makes code like this:
+```C
+extern double _Complex computes_function(double _Complex x);
+```
+callable in Rust without indirection:
+```rust
+extern "C" {
+  fn computes_function(x: Complex<f64>) -> Complex<f64>;
+}
+fn main() {
+  let returned_value = computes_function(Complex<f64>::new(3, 4))
+}
+```
+using the standard library's FFI-compatible complex numbers.
+
+## Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+`Complex<T>` numbers can be instantiated as of any type using `Complex::new(re, im)` where `re` and `im` are of the same type (this includes all numbers).
+```rust
+let x = Complex::new(3.0, 4.0);
+```
+They can even be passed as an array:
+```rust
+let y = Complex::from([3.0, 4.0]);
+```
+or as a tuple:
+```rust
+let z = Complex::from((3.0, 4.0));
+```
+They can even be passed in polar form (but only as a float):
+```rust
+let polar = Complex::from_polar(3.0, f32::PI/2.0);
+```
+where .i() turns a real number into a complex one transposing the real value to a complex value.
```

Comment:
The phrasing implies that `.i()` is used in the preceding example but it's not (this is the first mention that there's a thing called `.i()`)